graph_weather/models/atmorep/model/multiformer.py

```diff
@@ -0,0 +1,293 @@
+                  where E = num_ensemble_members.
+        """
+        # === 1) Encode each field with its Vision Transformer ===
+        encoded_features = {}
+        multi_scale_features = {}
+        for field_name, field_data in x_dict.items():
+            field_mask = mask_dict[field_name] if (mask_dict and field_name in mask_dict) else None
+            encoded, features = self.field_transformers[field_name](field_data, field_mask)
+            encoded_features[field_name] = encoded
+            multi_scale_features[field_name] = features
+
+        # === 2) Apply cross-attention among fields ===
+        cross_attended_features = {field: feat.clone() for field, feat in encoded_features.items()}
+        for f1 in self.input_fields:
+            for f2 in self.input_fields:
+                if f1 != f2:
+                    cross_key = f"{f2}_to_{f1}"
+                    cross_attn_layer = self.cross_attentions[cross_key]
+                    cross_attended_features[f1] += cross_attn_layer(
+                        encoded_features[f1], encoded_features[f2]
+                    )
+
+        # === 3) Decode each field using multi-scale features ===
+        decoded_features = {}
+        for field_name, feat in cross_attended_features.items():
+            decoded = self.decoders[field_name](feat, multi_scale_features[field_name])
+            decoded_features[field_name] = decoded
+
+        # === 4) Apply ensemble prediction heads ===
+        predictions = {}
+        for field_name, field_decoded in decoded_features.items():
+            B, T, N, D = field_decoded.shape
+            # Convert patch tokens back to 2D images of shape (H, W)
+            patch_grid = int(math.sqrt(N))
+            H = W = patch_grid * self.patch_size
+
+            field_preds = []
+            for head in self.ensemble_heads[field_name]:
+                # head: nn.Linear -> [B, T, N, D] -> [B, T, N, 1]
+                pred = head(field_decoded)
+
+                # Rearrange => [B, T, (H*C), W], then upsample
+                pred = rearrange(pred, "b t (h w) c -> b t (h c) w", h=patch_grid, w=patch_grid)
+                pred = F.interpolate(pred, size=(H, W), mode="bilinear")
+
+                # Rearrange => [B, T, W, H] if desired (or keep as channels)
+                # We'll treat last dimension as 'C' or 'Channels=1'
+                pred = rearrange(pred, "b t c w -> b t w c")
+                field_preds.append(pred)
+
+            # Stack across ensemble dimension => [E, B, T, H, W]
+            predictions[field_name] = torch.stack(field_preds, dim=0)
+
+        return predictions
+
+
+class EnhancedMultiFormer(MultiFormer):
+    """
+    EnhancedMultiFormer extends MultiFormer by adding spatiotemporal attention
+    after the regular transformer blocks to better capture temporal/spatial dependencies.
+
+    Args:
+        All the same args as MultiFormer, plus any needed for SpatioTemporalAttention.
```

Comment:
I would still copy the args over here, duplication in these kinds of docstrings is okay so that the documentation is right next to the code it is describing.